plugin/documents.py

```diff
@@ -216,6 +265,95 @@ def on_hover(self, point: int, hover_zone: int) -> None:
             return
         self.view.run_command("lsp_hover", {"point": point})
 
+    # --- textDocument/signatureHelp -----------------------------------------------------------------------------------
+
+    def _do_signature_help(self) -> None:
+        # NOTE: We take the beginning of the region to check the previous char (see last_char variable). This is for
+        # when a language server inserts a snippet completion.
+        pos = self._stored_region.a
+        if pos == -1:
+            return
+        if not self.view.match_selector(pos, self.view.settings().get("auto_complete_selector") or ""):  # ???
+            return
+        provider = None  # type: Optional[Dict[str, Any]]
+        session = None  # type: Optional[Session]
+        for sv in self.session_views_async():
+            provider = sv.session.get_capability("signatureHelpProvider")
+            if provider:
+                session = sv.session
+                break
+        if not provider or not session:
+            return
+        triggers = provider.get("triggerCharacters") or []
+        if not triggers:
+            return
+        last_char = self.view.substr(pos - 1)
+        if last_char in triggers:
+            self.purge_changes_async()
+            params = text_document_position_params(self.view, pos)
+            session.send_request(Request.signatureHelp(params), lambda resp: self._on_signature_help(resp, pos))
+        elif self.view.is_popup_visible():
+            if last_char.isspace():
+                # Peek behind to find the last non-whitespace character.
+                last_non_white_space_position = self.view.find_by_class(pos, False, ~0)
+                last_char = self.view.substr(last_non_white_space_position - 1)
+            if last_char not in triggers:
+                # TODO: Refactor popup usage to a common class. We now have sigHelp, completionDocs, hover, and diags
+                # all using a popup. Most of these systems assume they have exclusive access to a popup, while in
+                # reality there is only one popup per view.
+                self.view.hide_popup()
+
+    def _on_signature_help(self, response: Optional[Dict], point: int) -> None:
+        region = self._get_current_region_async()
+        if region is None:
+            return
+        if point != region.b:
+            return
```

Comment:
```js
callFn(arg1
```
Let's say I've typed `arg1` and I want to add another argument. I type `,[space]` and that makes signature popup not show up if I type those fast. I believe it's due to this point check that no longer holders.
I can reproduce it easily in typescript server in a bigger project than in pyls with LSP project.

![Screen Recording 2020-07-19 at 23 13 27](https://user-images.githubusercontent.com/153197/87885385-ca935a80-ca15-11ea-9bab-bc58a5414f78.gif)

Reply:
I see. Perhaps we can redo the checks earlier to see if it’s still applicable to show signature help.

Reply:
And by “checks earlier” I mean the checks that were done before sending the request.

Reply:
I think this is fixed in f34e7c3. The point check is unnecessary.